### PR TITLE
improve README documentation

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

Haivng just got a macOS build running (with lots of help from several people), I wanted to (hopefully) make it even clearer for those coming after me. I added a bunch to the readme. Please feel free to give feedback if some of my writing seems unclear, or my "tips" are actually things I just misunderstood.

I also got pedantic about the capitalization (or lack thereof) of `macOS`.

## Reference

No Jira ticket for this one.

## Checklist
    
- [N/A] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [N/A] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

**Comment by mcleinman** [comment]
Also please let me know if there are any pull request conventions/practices that I'm messing up!

**Review by bakulf — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -177,16 +203,25 @@ mozillavpn linuxdaemon needs privileged access and so if you do not run as
...
 vpn.
 
-### How to build from source code on MacOS
+### How to build from source code on macOS
+
+There are two ways to build the project on macOS - Through the terminal or with Xcode.
+For both, you must install a version of the VPN application in the main macOS Applications 
+folder so that the daemon is available for the debug build. Unless major changes were recently made, 
+you can likely use either the [latest release version](https://www.mozilla.org/en-US/products/vpn/download/) 
+or the [latest `main` branch build](https://firefox-ci-tc.services.mozilla.com/api/index/v1/task/mozillavpn.v2.mozilla-vpn-client.branch.main.latest.build.macos/artifacts/public%2Fbuild%2FMozillaVPN.pkg).
```
Comment (bakulf) [inline README.md:212]: Maybe we can link the latest official build instead: https://vpn.mozilla.org/r/vpn/download/mac

**Review by flozia — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mcleinman) [reply README.md:212]: That is essentially what the `latest release version` link on the above line represents - I'm going to drop this entire line, unless you think there is a better resolution.

Comment (mcleinman) [reply README.md]: Updated this

**Pull request merged**